### PR TITLE
fix(crashlytics)!: `checkForUnsentReports` should error if `isCrashlyticsCollectionEnabled` is false

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

This is a follow up PR for [FF-19](https://yt.invertase.io/youtrack/issue/FLUTTERFIRE-19) & [FF-20](https://yt.invertase.io/youtrack/issue/FLUTTERFIRE-20). 

Unsure if this is a breaking change as I've updated this [checkForUnsentReports()](https://github.com/FirebaseExtended/flutterfire/compare/@russell/crashlytics-follow-up-pr?expand=1#diff-ec378bc3ccbeedd0a391d573aacc9b6d96edbd7c82ab8bfc2993661151e8085aR43-R46) to throw an error if `isCrashlyticsCollectionEnabled` is set to `true`. This now matches the behaviour of [RNFB Crashlytics](https://github.com/invertase/react-native-firebase/blob/master/packages/crashlytics/lib/index.js#L52) and is the reason this [test](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_crashlytics/firebase_crashlytics/example/test_driver/firebase_crashlytics_e2e.dart#L22-L41) wasn't erroring as expected.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_crashlytics/firebase_crashlytics/example/test_driver/firebase_crashlytics_e2e.dart [hunk 1]
```diff
@@ -20,25 +20,17 @@ void testsMain() {
...
-      //   await expectLater(
-      //     crashlytics.checkForUnsentReports,
-      //     throwsA(isA<FirebaseException>()),
-      //   );
-      // });
+      test('should throw if automatic crash report is enabled', () async {
+        await crashlytics.setCrashlyticsCollectionEnabled(true);
+
+        try {
+          await crashlytics.checkForUnsentReports();
```
Comment (rrousselGit) [inline packages/firebase_crashlytics/firebase_crashlytics/example/test_driver/firebase_crashlytics_e2e.dart:27]: I don't underatand, why is the `expectLater` syntax not working?

**Review by rrousselGit — APPROVED** [review]

**Review by Salakar — CHANGES_REQUESTED** [review]
Updated the title to reflect this is actually a breaking change since it didn't throw previously.

**Review by Salakar — APPROVED** [review]

**Comment by Salakar** [comment]
Anything blocking shipping this? cc @rrousselGit / @russellwheatley - don't want PRs to go stale if they're good to go :)

**Comment by russellwheatley** [comment]
Nothing blocking from my perspective 👍

**Comment by rrousselGit** [comment]
Should be good to go!

**Pull request merged**